skimage/io/_plugins/matplotlib_plugin.py

```diff
@@ -169,10 +169,33 @@ def imshow_collection(ic, *args, **kwargs):
     """Display all images in the collection.
 
     """
-    fig, axes = plt.subplots(1, len(ic))
+    N = len(ic)
+    print N
+    k = (N * 12)**(0.5)
+    r1 = int(k/4)
+    r2 = r1 + 1
+    if N % r1 == 0:
+        c1 = N/r1
+    else:
+        c1 = N/r1 + 1
```

Comment:
Normally one-liners are to be avoided in Python, but I think in this case the code is too sparse and repetitive, so my suggestion would be:

```python
c1 = N/r1 if N % r1 == 0 else N/r1 + 1
c2 = N/r2 if N % r2 == 0 else N/r2 + 1
```

Alternatively,

```python
c1 = N/r1
if N % r1 == 0:
    c1 += 1
```

Either approach makes the symmetry between r1 and r2 more apparent.